build.gradle

```diff
@@ -68,28 +68,31 @@ jacocoTestReport {
 }
 
 dependencies {
-    compile 'org.slf4j:slf4j-api:1.7.26'
-    compile 'org.slf4j:slf4j-simple:1.7.26'
+    api 'org.slf4j:slf4j-api:1.7.26'
+    api 'org.slf4j:slf4j-simple:1.7.26'
 
-    compile('org.reflections:reflections:0.9.11') {
+    api('org.reflections:reflections:0.9.11') {
         exclude group: 'com.google.guava', module: 'guava'
     }
 
-    compile 'org.bouncycastle:bcprov-jdk15to18:1.64'
-    compile 'org.bouncycastle:bcpkix-jdk15to18:1.64'
-    compile 'org.json:json:20180813'
-    compile 'com.fasterxml.jackson.core:jackson-core:2.9.9'
-    compile 'com.fasterxml.jackson.core:jackson-databind:2.9.9'
-    compile 'com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.9.9'
-    compile('com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.9.9') {
+    api 'com.google.code.gson:gson:2.8.2'
+
+    api 'org.bouncycastle:bcprov-jdk15to18:1.64'
+    api 'org.bouncycastle:bcpkix-jdk15to18:1.64'
+    api 'org.json:json:20180813'
+    api 'com.fasterxml.jackson.core:jackson-core:2.9.9'
+    api 'com.fasterxml.jackson.core:jackson-databind:2.9.9'
+    api 'com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:2.9.9'
+    api('com.fasterxml.jackson.datatype:jackson-datatype-json-org:2.9.9') {
         // Use our JSON lib
         exclude group: 'org.json', module: 'json'
     }
-    compile('com.fasterxml.jackson.datatype:jackson-datatype-guava:2.9.9') {
+    api('com.fasterxml.jackson.datatype:jackson-datatype-guava:2.9.9') {
```

Comment:
`api` is transient, where `compile` is not (also, [`compile` is deprecated, so if we don't wanna use the transient `api`, we should migrate to `implementation`](https://stackoverflow.com/a/44419574/1311272)), If I understand it correctly, this will make sure that we are using the same dependency (in this case, the line you commented on: `guava`) across Java Common, Engine Library, Java Library and any code using Java Library (e.g. Android Wallet).

Reply:
^^ What he said, but transient -> transitive

Reply:
Ops, hehe, right! Transient is when a property is not saved in database!